internal/action/delete.go

```diff
@@ -21,48 +21,75 @@ func (s *Action) Delete(c *cli.Context) error {
 		return exit.Error(exit.Usage, nil, "Usage: %s rm name", s.Name)
 	}
 
-	if !recursive && s.Store.IsDir(ctx, name) && !s.Store.Exists(ctx, name) {
+	if recursive {
+		if len(c.Args().Tail()) > 1 {
+			return exit.Error(exit.Usage, nil, "Deleting multiple keys is not supported in recursive mode")
+		}
+
+		return s.deleteRecursive(ctx, name, c.Bool("force"))
+	}
+
+	if s.Store.IsDir(ctx, name) && !s.Store.Exists(ctx, name) {
 		return exit.Error(exit.Usage, nil, "Cannot remove %q: Is a directory. Use 'gopass rm -r %s' to delete", name, name)
 	}
 
 	// specifying a key is optional.
 	key := c.Args().Get(1)
 
-	if recursive && key != "" {
-		return exit.Error(exit.Usage, nil, "Can not use -r with a key. Invoke delete either with a key or with -r")
+	// multiple secrets, so not a key
+	if len(c.Args().Tail()) > 1 {
+		key = ""
+	}
+
+	names := append([]string{name}, c.Args().Tail()...)
+
+	if key != "" && s.Store.Exists(ctx, key) {
+		return exit.Error(exit.Unsupported, nil, "Ambiguous key %q, use 'gopass edit %s' to delete", key, name)
+	}
+
+	if !s.Store.Exists(ctx, name) {
```

Comment:
Shouldn't this be checking each item in `names` instead of just `name`?

Reply:
This will clash with the key deletion logic below. But the delete call will check that again for every secret.